src/main/java/com/wire/bots/roman/Sender.java

```diff
@@ -99,25 +96,93 @@ private UUID sendAudio(IncomingMessage message, UUID botId) throws Exception {
         try (WireClient wireClient = repo.getClient(botId)) {
             if (wireClient == null)
                 return null;
-            final byte[] bytes = base64Decode(message);
+
+            final Attachment attachment = message.attachment;
 
             final AudioPreview preview = new AudioPreview(
-                    message.attachment.filename,
-                    message.attachment.mimeType,
-                    message.attachment.duration,
-                    message.attachment.levels,
-                    bytes.length);
+                    attachment.name,
+                    attachment.mimeType,
+                    attachment.duration,
+                    attachment.levels,
+                    attachment.size.intValue());
+
+            wireClient.send(preview);
+
+            AssetBase asset;
+            if (message.attachment.meta != null) {
+                asset = new AudioAsset(preview.getMessageId(), attachment.mimeType);
+                setAssetMetadata(asset, message.attachment.meta);
+            } else if (message.attachment.data != null) {
+                final byte[] bytes = Base64.getDecoder().decode(message.attachment.data);
+                asset = new AudioAsset(bytes, preview);
+                uploadAssetData(wireClient, asset);
+            } else {
+                throw new Exception("Meta or Data need to be set");
+            }
+
+            wireClient.send(asset);
+            return asset.getMessageId();
+        }
+    }
+
+    @Nullable
+    private UUID sendAttachment(IncomingMessage message, UUID botId) throws Exception {
+        try (WireClient wireClient = repo.getClient(botId)) {
+            if (wireClient == null)
+                return null;
```

Comment:
maybe loging this? this means that Roman does not have client for the given bot, which is an exceptional state, is it not? 

that can happened when the bot was removed from the conversation but yet it tried to send the message, I think we should at least log such case --- this also applies to all cases that are here (not just assets)

Reply:
(maybe we log it in the upper layer of the code, not sure, just commenting here)

Reply:
Thanks. I log it now in getClientRepo()